feature/onboarding/src/main/java/com/napzak/market/onboarding/termsAgreement/TermsAgreementScreen.kt

```diff
@@ -0,0 +1,174 @@
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.vector.ImageVector
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.res.vectorResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.hilt.navigation.compose.hiltViewModel
+import com.napzak.market.designsystem.component.button.NapzakButton
+import com.napzak.market.designsystem.component.button.NapzakCheckedButton
+import com.napzak.market.designsystem.theme.NapzakMarketTheme
+import com.napzak.market.designsystem.R.drawable.ic_arrow_left
+import com.napzak.market.designsystem.R.drawable.ic_arrow_right
+import com.napzak.market.designsystem.R.drawable.ic_first_step_indicator
+import com.napzak.market.feature.onboarding.R.string.onboarding_all_agreement_title
+import com.napzak.market.feature.onboarding.R.string.onboarding_next
+import com.napzak.market.feature.onboarding.R.string.onboarding_title
+import com.napzak.market.onboarding.termsAgreement.model.TermType
+import com.napzak.market.onboarding.termsAgreement.model.TermsAgreementUiState
+import com.napzak.market.onboarding.termsAgreement.model.getDisplayLabel
+import com.napzak.market.util.common.openUrl
+
+@Composable
+fun TermsAgreementRoute(
+    onBackClick: () -> Unit,
+    onNextClick: () -> Unit,
+) {
+    val viewModel: TermsAgreementViewModel = hiltViewModel()
+    val uiState = viewModel.uiState.collectAsState().value
+
+    TermsAgreementScreen(
+        uiState = uiState,
+        onAllAgreementClick = viewModel::onAllAgreementClick,
+        onAgreementClick = viewModel::onAgreementClick,
+        onBackClick = onBackClick,
+        onNextClick = onNextClick,
+    )
+}
+
+@Composable
+fun TermsAgreementScreen(
+    uiState: TermsAgreementUiState,
+    onAllAgreementClick: () -> Unit,
+    onAgreementClick: (TermType) -> Unit,
+    onBackClick: () -> Unit,
+    onNextClick: () -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    val context = LocalContext.current
+
+    Column(
+        modifier = modifier
+            .fillMaxSize()
+            .background(NapzakMarketTheme.colors.white)
+            .padding(horizontal = 20.dp, vertical = 50.dp),
+    ) {
+        TermsAgreementTopBar(
+            onBackClick = onBackClick,
+        )
+
+        Spacer(modifier = Modifier.height(30.dp))
+
+        Text(
+            text = stringResource(onboarding_title),
+            style = NapzakMarketTheme.typography.title20b,
+            color = NapzakMarketTheme.colors.gray400,
+        )
+
+        Spacer(modifier = Modifier.height(24.dp))
+
+        NapzakCheckedButton(
+            checked = uiState.isAllAgreed,
+            text = stringResource(onboarding_all_agreement_title),
+            backgroundColor = NapzakMarketTheme.colors.gray50,
+            onCheckedChange = { onAllAgreementClick() },
```

Comment:
p1: 람다를 사용하지 않아도 됩니다! 람다를 사용하는 것도 객체를 생성하는 작업이기 때문에 가능하면 빼주도록 합시다!!
```suggestion
            onCheckedChange = onAllAgreementClick,
```

Reply:
해당 onCheckedChange는 (Boolean) -> Unit 타입을 요구하는데 onAllAgreementClick은 () -> Unit으로 파라미터가 없는 함수입니다! 따라서 시그니처가 달라 직접 참조는 불가능하고 불가피하게 람다로 감싸 전달해줘야 오류 없이 작동할 수 있을 것 같습니다! 람다 생성을 줄이는 방향은 맞지만 이 경우는 예외로 보는게 좋을 것 같아요!!

Reply:
아하 코드를 파악 못한 제 불찰이네요ㅎㅎ